data/tests/test_helpers.py

```diff
@@ -133,3 +133,136 @@ def test_bulk_check_spdis_in_arangodb():
+    assert result == {'variant_id': spdi, 'reason': 'Ambigious alt allele'}
+
+
+@patch('adapters.helpers.get_ref_seq_by_spdi', return_value='A')
+@patch('adapters.helpers.get_seqrepo')
+@patch('adapters.helpers.SeqRepoDataProxy')
+@patch('adapters.helpers.AlleleTranslator')
+@patch('adapters.helpers.build_spdi', return_value='NC_000001.11:12345:A:T')
+@patch('adapters.helpers.build_hgvs_from_spdi', return_value='NC_000001.11:g.12346A>T')
+def test_valid_spdi_input(mock_hgvs, mock_build, mock_translator, mock_proxy, mock_seqrepo, mock_ref_seq):
+    variant_id = 'NC_000001.11:12345:A:T'
+    result, skipped = load_variant(variant_id)
+    assert skipped is None
+    assert result['ref'] == 'A'
+    assert result['alt'] == 'T'
+    assert result['variation_type'] == 'SNP'
+    assert result['hgvs'] == 'NC_000001.11:g.12346A>T'
+
+
+def test_unable_to_parse_variant():
+    variant_id = 'BAD_FORMAT'
+    result, skipped = load_variant(variant_id)
+    assert result == {}
+    assert skipped == {'variant_id': variant_id,
+                       'reason': 'Unable to parse this variant id'}
+
+
+@patch('adapters.helpers.get_ref_seq_by_spdi', return_value='A')
+def test_empty_alt(mock_ref_seq):
+    variant_id = 'NC_000001.11:12345:A:'
+    result, skipped = load_variant(variant_id)
+    assert result['ref'] == 'A'
+    assert result['alt'] == ''
+    assert result['variation_type'] == 'deletion'
+    assert skipped is None or 'Ref allele mismatch' not in skipped.get(
+        'reason', '')
+
+
+@patch('adapters.helpers.get_ref_seq_by_spdi', return_value='T')
+def test_ref_mismatch_on_partial_spdi(mock_ref_seq):
+    variant_id = 'NC_000001.11:12345:A:'  # ref != genome
+    result, skipped = load_variant(variant_id)
+    assert result == {}
+    assert skipped['reason'] == 'Ref allele mismatch'
+
+
+@patch('adapters.helpers.get_seqrepo')
+@patch('adapters.helpers.SeqRepoDataProxy')
+@patch('adapters.helpers.AlleleTranslator')
+@patch('adapters.helpers.build_spdi', side_effect=ValueError('Ref allele mismatch'))
+def test_build_spdi_raises_error(mock_build, mock_translator, mock_proxy, mock_seqrepo):
+    variant_id = 'NC_000001.11:12345:A:T'
+    result, skipped = load_variant(variant_id)
+    assert result == {}
+    assert skipped['reason'] == 'Ref allele mismatch'
+
+
+@patch('adapters.helpers.get_seqrepo')
+@patch('adapters.helpers.SeqRepoDataProxy')
+@patch('adapters.helpers.AlleleTranslator')
+@patch('adapters.helpers.build_spdi', return_value='NC_000001.11:12345:A:N')
+def test_illegal_base_in_alt(mock_build, mock_translator, mock_proxy, mock_seqrepo):
+    variant_id = 'NC_000001.11:12345:A:N'
+    result, skipped = load_variant(variant_id)
+    assert result == {}
+    assert skipped == {'variant_id': variant_id,
+                       'reason': 'Ambigious alt allele'}
+
+
+@patch('adapters.helpers.get_ref_seq_by_spdi', return_value='T')
+def test_spdi_empty_ref(mock_ref_seq):
+    variant_id = 'NC_000010.11:79347444::CCTCCTCAGG'
```

Comment:
I think this should be accepted, return_value for get_ref_seq_by_spdi should be ''

Reply:
fixed